Cloud user investigation (#28843)
demisto/content: Packs/Azure-Enrichment-Remediation/Playbooks/playbook-Azure_-_User_Investigation_README.md

This playbook performs an investigation on a specific user in Azure environments, using queries and logs from Azure Log Analytics.

## Dependencies

This playbook uses the following sub-playbooks, integrations, and scripts.

### Sub-playbooks

This playbook does not use any sub-playbooks.

### Integrations

> * Azure Log Analytics

Reviewer: ```suggestion
Azure Log Analytics
```

### Scripts

* SetMultipleValues
* SetAndHandleEmpty

### Commands

> * azure-log-analytics-execute-query

Reviewer: ```suggestion
azure-log-analytics-execute-query
```

## Playbook Inputs

---

| **Name** | **Description** | **Default Value** | **Required** |
| --- | --- | --- | --- |
| Username | The username to investigate. | avishai@demistodev.onmicrosoft.com | Optional |
| AzureSearchTime | The Search Time for the Azure Log Analytics search query. Default value: ago\(1d\) | ago(7d) | Optional |

> | failedLogonThreshold | The threshold number of failed login by the user. required to determine how many failed logon event count as suspicious events. | 20 | Optional |

Reviewer: ```suggestion
| failedLogonThreshold | The threshold number of failed logins by the user. Required to determine how many failed logon events count as suspicious events. | 20 | Optional |
```

> | MfaAttemptThreshold | The threshold number of MFA failed login by the user. required to determine how many MFA failed logon event count as suspicious events. | 10 | Optional |

Reviewer: ```suggestion
| MfaAttemptThreshold | The threshold number of MFA failed logons by the user. Required to determine how many MFA failed logon events count as suspicious events. | 10 | Optional |
```

## Playbook Outputs

---

| **Path** | **Description** | **Type** |
| --- | --- | --- |
| AzureScriptBasedUserAgentEvents | Script-based user agent events used by the user in the Azure environment. | unknown |

> | CountAzureEvents.AzureScriptBasedUserAgentCount | Count of script-based user agent usages by the user in the Azure environment. | unknown |

Reviewer: ```suggestion
| CountAzureEvents.AzureScriptBasedUserAgentCount | Number of script-based user agent usages by the user in the Azure environment. | unknown |
```

| AzureAdminActivitiesEvents | Administrative activities performed by the user in the Azure environment. | unknown |

> | CountAzureEvents.AzureAdminActivitiesCount | Count of administrative activities performed by the user in the Azure environment. | unknown |

Reviewer: ```suggestion
| CountAzureEvents.AzureAdminActivitiesCount | Number of administrative activities performed by the user in the Azure environment. | unknown |
```

| AzureSecurityRulesChangeEvents | Security rules that were changed by the user in the Azure environment. | unknown |

> | CountAzureEvents.AzureSecurityRulesChangeCount | Count of the security rules that were changed by the user in the Azure environment. | unknown |

Reviewer: ```suggestion
| CountAzureEvents.AzureSecurityRulesChangeCount | Number of security rules that were changed by the user in the Azure environment. | unknown |
```

> | AzureUnsuccessSecurityRulesChangeEvents | An unsuccessful attempts to change security rules by the user in the Azure environment. | unknown |

Reviewer: ```suggestion
| AzureUnsuccessSecurityRulesChangeEvents | Unsuccessful attempts to change security rules by the user in the Azure environment. | unknown |
```

> | CountAzureEvents.AzureUnsuccessSecurityRulesChangeCount | count of unsuccessful attempts to change security rules by the user in the Azure environment. | unknown |

Reviewer: ```suggestion
| CountAzureEvents.AzureUnsuccessSecurityRulesChangeCount | Number of unsuccessful attempts to change security rules by the user in the Azure environment. | unknown |
```

> | AzureFailLoginCount | Count of failed logins by the user in the Azure environment. | unknown |

Reviewer: ```suggestion
| AzureFailLoginCount | Number of failed logins by the user in the Azure environment. | unknown |
```

> | AzureFailLoginMFACount | Count of failed logins by the user using MFA in the Azure environment. | unknown |

Reviewer: ```suggestion
| AzureFailLoginMFACount | Number of failed logins by the user using MFA in the Azure environment. | unknown |
```

> | AzureAnomaliesEvents | Anomalies Events on the user in the Azure environment. | unknown |

Reviewer: ```suggestion
| AzureAnomaliesEvents | Anomaly events on the user in the Azure environment. | unknown |
```

> | CountAzureEvents.AzureAnomaliesCount | Count of anomalies Events on the user in the Azure environment. | unknown |

Reviewer: ```suggestion
| CountAzureEvents.AzureAnomaliesCount | Number of anomaly events on the user in the Azure environment. | unknown |
```

> | AzureRiskyUserCount | Count the events where the user was defined as a risky user  in the Azure environment. | unknown |

Reviewer: ```suggestion
| AzureRiskyUserCount | Number of events where the user was defined as a risky user  in the Azure environment. | unknown |
```

> | AzureRiskyUserCount | Count the events where the user was defined as a risky user  in the Azure environment. | unknown |

Reviewer: ```suggestion
| AzureRiskyUserCount | The number of events where the user was defined as a risky user in the Azure environment. | unknown |
```

| AzureUncommonCountryLogonEvents | Uncommon country logon events by the user in the Azure environment. | unknown |

> | CountAzureEvents.AzureUncommonCountryLogonCount | Count of uncommon country logon events by the user in the Azure environment. | unknown |

Reviewer: ```suggestion
| CountAzureEvents.AzureUncommonCountryLogonCount | Number of uncommon country logon events by the user in the Azure environment. | unknown |
```

| AzureUncommonVolumeEvents | Uncommon volume events by the user in the Azure environment. | unknown |

> | CountAzureEvents.AzureUncommonVolumeCount | Count of uncommon volume events by the user in the Azure environment. | unknown |

Reviewer: ```suggestion
| CountAzureEvents.AzureUncommonVolumeCount | Number of uncommon volume events by the user in the Azure environment. | unknown |
```

> | AzureUncommonActivitiesEvents | Uncommon activities events by the user in the Azure environment. | unknown |

Reviewer: ```suggestion
| AzureUncommonActivitiesEvents | Uncommon activity events by the user in the Azure environment. | unknown |
```

> | CountAzureEvents.AzureUncommonActivitiesCount | Count of uncommon activities events by the user in the Azure environment. | unknown |

Reviewer: ```suggestion
| CountAzureEvents.AzureUncommonActivitiesCount | Number of uncommon activity events by the user in the Azure environment. | unknown |
```

This playbook performs an investigation on a specific user in Azure environments, using queries and logs from Azure Log Analytics to locate the following activities performed by the user:
- Script-based user agent usage
- Administrative user activities
- Security rules and policies changes
- Failed login attempt
- MFA failed login attempt
- Login attempt from an uncommon country.
- Anomalies activities
- Risky users
- Uncommon high volume of actions
- Action uncommonly performed by the user

## Dependencies

This playbook uses the following sub-playbooks, integrations, and scripts.

### Sub-playbooks

This playbook does not use any sub-playbooks.

### Integrations

> * Azure Log Analytics

Reviewer: ```suggestion
Azure Log Analytics
```

### Scripts

* SetMultipleValues
* SetAndHandleEmpty

### Commands

> * azure-log-analytics-execute-query

Reviewer: ```suggestion
azure-log-analytics-execute-query
```
